File: cafeoccidente-backend/src/main/java/com/cafeoccidente/backend/controlrecord/service/ControlRecordService.java
package com.cafeoccidente.backend.controlrecord.service;

import com.cafeoccidente.backend.controlrecord.dto.ControlRecordRequest;
import com.cafeoccidente.backend.controlrecord.dto.ControlRecordResponse;
import com.cafeoccidente.backend.controlrecord.entity.ControlRecord;
import java.util.List;

/** Unica responsabilidad: exponer y administrar los parametros generales (RegControl) de cada agencia. */
public interface ControlRecordService {
    ControlRecord getActive(Long agencyId);

    List<ControlRecordResponse> list();

    ControlRecordResponse getByAgency(Long agencyId);

    ControlRecordResponse create(Long agencyId, ControlRecordRequest request);

    ControlRecordResponse update(Long agencyId, ControlRecordRequest request);
}

File: cafe-occidente-frontend/src/app/core/services/control-record.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { API_BASE_URL } from '../config/api.config';
import { ControlRecordRequest, ControlRecordResponse } from '../models/control-record.model';

/** Unica responsabilidad: llamadas HTTP al backend para el recurso ControlRecord (RegControl), por agencia. */
@Injectable({ providedIn: 'root' })
export class ControlRecordService {
  private readonly http = inject(HttpClient);

  getByAgency(agencyId: number): Observable<ControlRecordResponse> {
    return this.http.get<ControlRecordResponse>(`${API_BASE_URL}/control-records/${agencyId}`);
  }

  create(agencyId: number, request: ControlRecordRequest): Observable<ControlRecordResponse> {
    return this.http.post<ControlRecordResponse>(`${API_BASE_URL}/control-records/${agencyId}`, request);
  }

  update(agencyId: number, request: ControlRecordRequest): Observable<ControlRecordResponse> {
    return this.http.put<ControlRecordResponse>(`${API_BASE_URL}/control-records/${agencyId}`, request);
  }
}
